portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,46 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  ArrayList<String> units=new ArrayList<String>();
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    units.add("feet");
+    units.add("inches");
+    units.add("meters");
+    units.add("miles");
+    units.add("pounds");
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.getWriter().println("Hello Olu!");
+    String json = convertToJson(units);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  private String convertToJson(ArrayList<String> units) {
```

Comment:
I think this code will be easier to write and maintain if you use a library to do it.  The documentation suggests using Gson - you can see an example of that here: https://github.com/googleinterns/step/blob/master/walkthroughs/week-3-server/examples/server-stats/src/main/java/com/google/sps/servlets/ServerStatsServlet.java

You will probably need to add a line like `import com.google.gson.Gson;` at the top, then you can use a function like this: https://github.com/googleinterns/step/blob/master/walkthroughs/week-3-server/examples/server-stats/src/main/java/com/google/sps/servlets/ServerStatsServlet.java#L71

Reply:
Addressed in 3.3changes